examples/c/ex_key_management.c

```diff
@@ -0,0 +1,175 @@
+
+    return (0);
+}
+
+/*
+ * my_set_key_blob --
+ *     A placeholder example of set_key_blob() call.
+ */
+static int
+my_set_key_blob(WT_KEY_MANAGEMENT *km, void *blob_data, size_t blob_size, bool has_changed)
+{
+    MY_KEY_MANAGEMENT *my_km = (MY_KEY_MANAGEMENT *)km;
+
+    WT_UNUSED(blob_data);
+    WT_UNUSED(blob_size);
+    WT_UNUSED(my_km);
+
+    WT_UNUSED(has_changed);
+
+    return (0);
+}
+
+/*
+ * my_set_key_blob_completed --
+ *     A placeholder example of set_key_complete() call.
+ */
+static int
+my_set_key_blob_completed(WT_KEY_MANAGEMENT *km, void *callback)
+{
+    MY_KEY_MANAGEMENT *my_km = (MY_KEY_MANAGEMENT *)km;
+
+    WT_UNUSED(my_km);
+    WT_UNUSED(callback);
+
+    return (0);
+}
+
+/*
+ * set_my_key_management --
+ *     A simple example of setting the key management system.
+ */
+int
+set_my_key_management(WT_CONNECTION *conn, WT_CONFIG_ARG *config)
+{
+    MY_KEY_MANAGEMENT *kms;
+    WT_KEY_MANAGEMENT *wt;
+    WT_EXTENSION_API *wtext;
+
+    WT_UNUSED(config);
+    wtext = conn->get_extension_api(conn);
+    /*
+     * Initialize our key management system.
+     */
+    if ((kms = calloc(1, sizeof(MY_KEY_MANAGEMENT))) == NULL) {
+        (void)wtext->err_printf(
+          wtext, NULL, "set_my_key_management: %s", wtext->strerror(wtext, NULL, ENOMEM));
+        return (errno);
+    }
+    wt = (WT_KEY_MANAGEMENT *)&kms->km;
+    wt->get_key_blob = my_get_key_blob;
+    wt->set_key_blob = my_set_key_blob;
+    wt->set_key_complete = my_set_key_blob_completed;
+
+    kms->key_blob.id = 1;
+    kms->key_blob.data = 1234;
+    error_check(conn->set_key_management(conn, (WT_KEY_MANAGEMENT *)kms, NULL));
+    return (0);
+}
+
+static const char *home;
+
+int
+main(int argc, char *argv[])
+{
+    WT_CONNECTION *conn;
+    const char *open_config;
+    int ret = 0;
+
+    WT_UNUSED(argc);
+    WT_UNUSED(argv);
+
+    /*
+     * Create a clean test directory for this run of the test program if the environment variable
+     * isn't already set (as is done by make check).
+     */
+    if (getenv("WIREDTIGER_HOME") == NULL) {
+        home = "WT_HOME";
+        ret = system("rm -rf WT_HOME && mkdir WT_HOME");
+    } else
+        home = NULL;
+
+    /*! [WT_KEY_MANAGEMENT register] */
+    /*
+     * Setup a configuration string that will load our key management system. Use the special local
+     * extension to indicate that the entry point is in the same executable. Also enable early load
+     * for this extension, since WiredTiger needs to be able to find it before doing any operations.
+     */
+    open_config =
+      "create,log=(enabled=true),extensions=(local={entry=set_my_key_management,early_load=true})";
+    /* Open a connection to the database, creating it if necessary. */
+    if ((ret = wiredtiger_open(home, NULL, open_config, &conn)) != 0) {
+        fprintf(stderr, "Error connecting to %s: %s\n", home == NULL ? "." : home,
+          wiredtiger_strerror(ret));
+        return (EXIT_FAILURE);
+    }
```

Comment:
This example is just steps away from being a test program...  If it were to create a table, put something in it (so the checkpoint is not empty), then checkpoint, and then reopen the connection, it would cause all the new machinery in the API to be executed.  I'm not saying you should add this to the test program - but it would make a nice starting place for a C test.